mesheryctl/pkg/utils/helpers.go

```diff
@@ -592,6 +601,50 @@ func ValidateURL(URL string) error {
 	return nil
 }
 
+// ListManifests lists the manifest files stored in GitHub
+func ListManifests(url string) ([]Manifest, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to make GET request to %s", url)
+	}
+	defer SafeClose(resp.Body)
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read response body")
+	}
+
+	var manLis ManifestList
+
+	json.Unmarshal([]byte(body), &manLis)
+	return manLis.Tree, nil
+}
+
+// GetManifestURL returns the URLs for the manifest files
+func GetManifestURL(manifest Manifest, rawManifestsURL string) string {
+	var manifestURL string
+
+	if manifest.Typ == "blob" {
+		manifestURL = rawManifestsURL + manifest.Path
+		return manifestURL
+	}
+	return ""
+}
+
+// DownloadManifests downloads all the Kubernetes manifest files
+func DownloadManifests(manifestArr []Manifest, rawManifestsURL string) error {
+	for _, manifest := range manifestArr {
+		if manifestFile := GetManifestURL(manifest, rawManifestsURL); manifestFile != "" {
+			// download the manifest files to ~/.meshery/manifests folder
+			filepath := filepath.Join(MesheryFolder, ManifestsFolder, manifest.Path)
+			if err := DownloadFile(filepath, manifestFile); err != nil {
```

Comment:
`meshkit` has also a `Download` function, maybe start using that one (if it fits) to eventually remove the corresponding code in this repo?

Reply:
Yes. That seems like a good idea. Since we have not imported MeshKit until now, we have a local `DownloadFile` function. Switching to that makes sense. Will need to also look for any other functions that have been re-implemented.